plugin-base/src/main/java/io/everytrade/server/plugin/impl/everytrade/parser/exchange/bean/BitstampBeanV1.java

```diff
@@ -55,40 +68,62 @@ public void setAmount(String amount) {
 
     @Parsed(field = "Value")
     public void setValue(String value) {
-        String[] valueParts = value.split(" ");
-        valueCurrency = Currency.fromCode(valueParts[1]);
-        if (valueParts[0].isEmpty()) {
+        if (value == null) {
             this.value = BigDecimal.ZERO;
         } else {
+            String[] valueParts = value.split(" ");
             this.value = new BigDecimal(valueParts[0]);
+            valueCurrency = Currency.fromCode(valueParts[1]);
         }
     }
 
     @Parsed(field = "Rate")
     public void setRate(String rate) {
-        rateCurrency = Currency.fromCode(rate.split(" ")[1]);
+        if (rate != null) {
+            String[] r = rate.split(" ");
+            rateCurrency = Currency.fromCode(r[1]);
+        }
     }
 
     @Parsed(field = "Fee")
     public void setFee(String fee) {
-        String[] feeParts = fee.split(" ");
-        if (feeParts[0].isEmpty()) {
+        if (fee == null) {
             this.fee = BigDecimal.ZERO;
         } else {
+            String[] feeParts = fee.split(" ");
             this.fee = new BigDecimal(feeParts[0]);
+            feeCurrency = Currency.fromCode(feeParts[1]);
         }
-        feeCurrency = Currency.fromCode(feeParts[1]);
     }
 
     @Parsed(field = "Sub Type")
     public void setSubType(String field) {
-        subType = detectTransactionType(field);
+        if (field != null) {
+            subType = detectTransactionType(field);
+        } else if (type.equals("Deposit") || type.equals("Withdrawal")) {
+            subType = detectTransactionType(type);
```

Comment:
Are you sure about "type" variable? Probably ok. Just asking.

Reply:
Yes, I am.